Add CI for no_std compilation and tests

As discussed in #23, we need to be sure that this continues to work.

I assume this can be done by running `cargo check --no-default-features` on CI, which currently yields 8 errors:

```commandline
$ cargo check --no-default-features --message-format short
    Checking stretch2 v0.4.2 (/home/tim/Documents/stretch)
src/sys.rs:62:35: error[E0433]: failed to resolve: could not find `consts` in `heapless`
src/sys.rs:63:36: error[E0433]: failed to resolve: could not find `consts` in `heapless`
src/sys.rs:64:38: error[E0433]: failed to resolve: could not find `consts` in `heapless`
src/sys.rs:80:24: error[E0405]: cannot find trait `Array` in crate `arrayvec`
src/sys.rs:66:56: error[E0747]: type provided when a constant was expected
src/sys.rs:67:35: error[E0107]: this struct takes 2 generic arguments but 1 generic argument was supplied
src/sys.rs:68:43: error[E0107]: this struct takes 2 generic arguments but 1 generic argument was supplied
src/sys.rs:69:42: error[E0107]: this struct takes 2 generic arguments but 1 generic argument was supplied
error: could not compile `stretch2` due to 8 previous errors
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -17,6 +17,21 @@ jobs:
         with:
           command: check
 
+  check-no-default-features:
+    name: Check (no default features)
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: check
+          args: --no-default-features
+
   test:
     name: Test Suite
     runs-on: ubuntu-latest
@@ -30,6 +45,21 @@ jobs:
       - uses: actions-rs/cargo@v1
         with:
           command: test
+
+  test-no-default-features:
+    name: Test Suite (no default features)
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --no-default-features
 
   fmt:
     name: Rustfmt

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -3,50 +3,6 @@ on: [push, pull_request]
 name: Continuous integration
 
 jobs:
-  check-features-none:
-    name: "Check [Features: None]"
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: stable
-          override: true
-      - uses: actions-rs/cargo@v1
-        with:
-          command: check
-          args: --no-default-features
-
-  check-features-default:
-    name: "Check [Features: Default]"
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: stable
-          override: true
-      - uses: actions-rs/cargo@v1
-        with:
-          command: check
-
-  check-features-alloc:
-    name: "Check [Features: alloc]"
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: stable
-          override: true
-      - uses: actions-rs/cargo@v1
-        with:
-          command: check
-          args: --no-default-features --features alloc
-
   test-features-none:
     name: "Test Suite [Features: None]"
     runs-on: ubuntu-latest

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -3,22 +3,8 @@ on: [push, pull_request]
 name: Continuous integration
 
 jobs:
-  check:
-    name: Check
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: stable
-          override: true
-      - uses: actions-rs/cargo@v1
-        with:
-          command: check
-
-  check-no-default-features:
-    name: Check (no default features)
+  check-features-none:
+    name: "Check [Features: None]"
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
@@ -32,8 +18,8 @@ jobs:
           command: check
           args: --no-default-features
 
-  test:
-    name: Test Suite
+  check-features-default:
+    name: "Check [Features: Default]"
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
@@ -44,10 +30,25 @@ jobs:
           override: true
       - uses: actions-rs/cargo@v1
         with:
-          command: test
+          command: check
 
-  test-no-default-features:
-    name: Test Suite (no default features)
+  check-features-alloc:
+    name: "Check [Features: alloc]"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: check
+          args: --no-default-features --features alloc
+
+  test-features-none:
+    name: "Test Suite [Features: None]"
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
@@ -60,6 +61,35 @@ jobs:
         with:
           command: test
           args: --no-default-features
+
+  test-features-default:
+    name: "Test Suite [Features: Default]"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: test
+
+  test-features-alloc:
+    name: "Test Suite [Features: alloc]"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --no-default-features --features alloc
 
   fmt:
     name: Rustfmt

--- a/src/sys.rs
+++ b/src/sys.rs
@@ -57,16 +57,14 @@ mod alloc {
 
 #[cfg(all(not(feature = "alloc"), not(feature = "std")))]
 mod core {
-    use typenum::marker_traits::Unsigned;
+    const MAX_NODE_COUNT: usize = 256;
+    const MAX_CHILD_COUNT: usize = 16;
+    const MAX_PARENTS_COUNT: usize = 1;
 
-    type MaxNodeCount = heapless::consts::U256;
-    type MaxChildCount = heapless::consts::U16;
-    type MaxParentsCount = heapless::consts::U1;
-
-    pub type Map<K, V> = ::heapless::FnvIndexMap<K, V, MaxNodeCount>;
-    pub type Vec<A> = ::arrayvec::ArrayVec<[A; MaxNodeCount::USIZE]>;
-    pub type ChildrenVec<A> = ::arrayvec::ArrayVec<[A; MaxChildCount::USIZE]>;
-    pub type ParentsVec<A> = ::arrayvec::ArrayVec<[A; MaxParentsCount::USIZE]>;
+    pub type Map<K, V> = ::heapless::FnvIndexMap<K, V, MAX_NODE_COUNT>;
+    pub type Vec<A> = ::arrayvec::ArrayVec<[A; MAX_NODE_COUNT]>;
+    pub type ChildrenVec<A> = ::arrayvec::ArrayVec<[A; MAX_CHILD_COUNT]>;
+    pub type ParentsVec<A> = ::arrayvec::ArrayVec<[A; MAX_PARENTS_COUNT]>;
 
     pub fn new_map_with_capacity<K, V>(_capacity: usize) -> Map<K, V>
     where

--- a/src/sys.rs
+++ b/src/sys.rs
@@ -62,9 +62,9 @@ mod core {
     const MAX_PARENTS_COUNT: usize = 1;
 
     pub type Map<K, V> = ::heapless::FnvIndexMap<K, V, MAX_NODE_COUNT>;
-    pub type Vec<A> = ::arrayvec::ArrayVec<[A; MAX_NODE_COUNT]>;
-    pub type ChildrenVec<A> = ::arrayvec::ArrayVec<[A; MAX_CHILD_COUNT]>;
-    pub type ParentsVec<A> = ::arrayvec::ArrayVec<[A; MAX_PARENTS_COUNT]>;
+    pub type Vec<A> = ::arrayvec::ArrayVec<A, MAX_NODE_COUNT>;
+    pub type ChildrenVec<A> = ::arrayvec::ArrayVec<A, MAX_CHILD_COUNT>;
+    pub type ParentsVec<A> = ::arrayvec::ArrayVec<A, MAX_PARENTS_COUNT>;
 
     pub fn new_map_with_capacity<K, V>(_capacity: usize) -> Map<K, V>
     where

--- a/src/sys.rs
+++ b/src/sys.rs
@@ -73,10 +73,7 @@ mod core {
         Map::new()
     }
 
-    pub fn new_vec_with_capacity<T, A>(_capacity: usize) -> ::arrayvec::ArrayVec<A>
-    where
-        A: ::arrayvec::Array<Item = T>,
-    {
+    pub fn new_vec_with_capacity<A, const CAP: usize>(_capacity: usize) -> ::arrayvec::ArrayVec<A, CAP> {
         ::arrayvec::ArrayVec::new()
     }
 

